clang/test/CIR/CodeGen/applearm64-array-cookies.cpp

```diff
@@ -0,0 +1,64 @@
+// RUN: %clang_cc1 -std=c++20 -triple=arm64e-apple-darwin -fclangir -emit-cir %s -o %t.cir
+// RUN: FileCheck --input-file=%t.cir %s
+
+class C {
+  public:
+    ~C();
+};
+
+void t_constant_size_nontrivial() {
+  auto p = new C[3];
+}
+
+// CHECK:  cir.func @_Z26t_constant_size_nontrivialv()
+// CHECK:    %0 = cir.alloca !cir.ptr<!ty_C>, !cir.ptr<!cir.ptr<!ty_C>>, ["p", init] {alignment = 8 : i64}
+// CHECK:    %[[#NUM_ELEMENTS:]] = cir.const #cir.int<3> : !u64i
+// CHECK:    %[[#SIZE_WITHOUT_COOKIE:]] = cir.const #cir.int<3> : !u64i
+// CHECK:    %[[#ALLOCATION_SIZE:]] = cir.const #cir.int<19> : !u64i
+// CHECK:    %4 = cir.call @_Znam(%[[#ALLOCATION_SIZE]]) : (!u64i) -> !cir.ptr<!void>
```

Comment:
For SSA value that are not relevant there are few options: (a) use `{{.*}}`, (b) omit them if possible (c) remove the line altogether.

Reply:
I don't know how much detail needs to be captured here. All of it seems relevant if we want to track generated IR stability. Given how small the input function is, it shouldn't change much.

Reply:
I think you've captured enough, I usually stay away of the literal values because small changes in MLIR or in CIR operations might force us to fix unrelated testcases. You could still trim off a bit unrelated bits while keep checking for the operations you expect, example:
```
...
// CHECK: cir.ptr_stride({{.*}} %[[#COOKIE_SIZE]]
```

Reply:
As I was developing this I was interested in understanding why all the various bitcasts and stores were there, but I don't think it's necessarily important to check for all that. I'll trim it back a bit.